### ISSUE TITLE
nushell panicing and requires killing it

### ISSUE THREAD

**Issue by fdncred** [nushell/nushell#13118]
### Describe the bug

This morning, on WSL, I rebuilt nushell on the latest main and all the plugins. I launch nushell with and I get endless pages of nushell crashing, then recovering, then crashing again.

### Without backtrace, just running `nu`
```
Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

### With RUST_BACKTRACE=1 in release
```
Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
         0: 0x55fb9f4b86e4 - rust_begin_unwind
         1: 0x55fb9e69b2a5 - core::panicking::panic_fmt::ha6effc2775a0749c
         2: 0x55fb9e69b263 - core::option::expect_failed::hea24986454718b4f
         3: 0x55fb9f1dae65 - <&nu_protocol::engine::engine_state::EngineState as nu_protocol::span::GetSpan>::get_span::hdbbce8db39d1c6ed
         4: 0x55fb9ede46c4 - nu_protocol::eval_base::Eval::eval::hbffe37539539151d
         5: 0x55fb9edeec15 - nu_engine::eval::eval_expression::h90bd0635be8b1527
         6: 0x55fb9edef7e7 - nu_engine::eval::eval_expression_with_input::h4c8fe129b5520131
         7: 0x55fb9edf10cd - nu_engine::eval::eval_element_with_input_inner::h9d55357f53506491
         8: 0x55fb9edf1631 - nu_engine::eval::eval_element_with_input::h8b79302f84901d47
         9: 0x55fb9edf2a91 - nu_engine::eval::eval_block::h8e49bde5577b556b
        10: 0x55fb9e8c4548 - nu_cli::reedline_config::add_menus::h34b6b871574ccc16
        11: 0x55fb9e90f1ef - nu_cli::repl::loop_iteration::h13f55a99c51cb8e8
        12: 0x55fb9e909100 - std::panicking::try::ha8f8f5711fed4b7d
        13: 0x55fb9e8fd8ca - nu_cli::repl::evaluate_repl::h4f993c0b361316d3
        14: 0x55fb9e895c62 - nu::run::run_repl::h9f620c4636f0f651
        15: 0x55fb9e881888 - nu::main::hbb1caa39dfab20b1
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
         0: 0x55fb9f4b86e4 - rust_begin_unwind
         1: 0x55fb9e69b2a5 - core::panicking::panic_fmt::ha6effc2775a0749c
         2: 0x55fb9e69b263 - core::option::expect_failed::hea24986454718b4f
         3: 0x55fb9f1dae65 - <&nu_protocol::engine::engine_state::EngineState as nu_protocol::span::GetSpan>::get_span::hdbbce8db39d1c6ed
         4: 0x55fb9ede46c4 - nu_protocol::eval_base::Eval::eval::hbffe37539539151d
         5: 0x55fb9edeec15 - nu_engine::eval::eval_expression::h90bd0635be8b1527
         6: 0x55fb9edef7e7 - nu_engine::eval::eval_expression_with_input::h4c8fe129b5520131
         7: 0x55fb9edf10cd - nu_engine::eval::eval_element_with_input_inner::h9d55357f53506491
         8: 0x55fb9edf1631 - nu_engine::eval::eval_element_with_input::h8b79302f84901d47
         9: 0x55fb9edf2a91 - nu_engine::eval::eval_block::h8e49bde5577b556b
        10: 0x55fb9e8c4548 - nu_cli::reedline_config::add_menus::h34b6b871574ccc16
        11: 0x55fb9e90f1ef - nu_cli::repl::loop_iteration::h13f55a99c51cb8e8
        12: 0x55fb9e909100 - std::panicking::try::ha8f8f5711fed4b7d
        13: 0x55fb9e8fd8ca - nu_cli::repl::evaluate_repl::h4f993c0b361316d3
        14: 0x55fb9e895c62 - nu::run::run_repl::h9f620c4636f0f651
        15: 0x55fb9e881888 - nu::main::hbb1caa39dfab20b1
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
zsh: terminated  RUST_BACKTRACE=1 nu
```

### With RUST_BACKTRACE=full in debug

```
Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
  ╰─▶ internal error: missing span
         0: 0x55f43366b264 - rust_begin_unwind
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/std/src/panicking.rs:647
         1: 0x55f43015a7c5 - core::panicking::panic_fmt::ha6effc2775a0749c
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/panicking.rs:72
         2: 0x55f43015a783 - core::panicking::panic_display::h89227dd7f9f02208
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/panicking.rs:196
                       - core::panicking::panic_str::h9477b77ceb5900f1
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/panicking.rs:171
                       - core::option::expect_failed::hea24986454718b4f
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/option.rs:1988
         3: 0x55f432ac8122 - core::option::Option<T>::expect::h1fcc5041a925dde1
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/option.rs:894
         4: 0x55f4329af1c4 - <&nu_protocol::engine::engine_state::EngineState as nu_protocol::span::GetSpan>::get_span::h6ddb8f5a3d91ebd0
                      at /home/fdncred/src/nushell/crates/nu-protocol/src/engine/engine_state.rs:1047
         5: 0x55f4328eef26 - nu_protocol::ast::expression::Expression::span::h3d010da045b5b455
                      at /home/fdncred/src/nushell/crates/nu-protocol/src/ast/expression.rs:520
         6: 0x55f4328c7178 - nu_protocol::eval_base::Eval::eval::h0db1bd3f6a6d871c
                      at /home/fdncred/src/nushell/crates/nu-protocol/src/eval_base.rs:26
         7: 0x55f4328d944a - nu_engine::eval::eval_expression::h0fbbac5ac797413b
                      at /home/fdncred/src/nushell/crates/nu-engine/src/eval.rs:237
         8: 0x55f4328d9cd4 - nu_engine::eval::eval_expression_with_input::h1fbfc971832f5c03
                      at /home/fdncred/src/nushell/crates/nu-engine/src/eval.rs:286
         9: 0x55f4328e01e2 - nu_engine::eval::eval_element_with_input_inner::hea7b8ecc13c68728
                      at /home/fdncred/src/nushell/crates/nu-engine/src/eval.rs:400
        10: 0x55f4328e117a - nu_engine::eval::eval_element_with_input::h5083d9b130f08454
                      at /home/fdncred/src/nushell/crates/nu-engine/src/eval.rs:480
        11: 0x55f4328e5bb9 - nu_engine::eval::eval_block::h7f114de64b95efad
                      at /home/fdncred/src/nushell/crates/nu-engine/src/eval.rs:552
        12: 0x55f43192374c - nu_cli::reedline_config::add_menus::h82a82d7c50cbda7f
                      at /home/fdncred/src/nushell/crates/nu-cli/src/reedline_config.rs:116
        13: 0x55f4318d5764 - nu_cli::repl::loop_iteration::hab334e15b963a40d
                      at /home/fdncred/src/nushell/crates/nu-cli/src/repl.rs:431
        14: 0x55f431881682 - nu_cli::repl::evaluate_repl::{{closure}}::h6456cf2b0fde6fda
                      at /home/fdncred/src/nushell/crates/nu-cli/src/repl.rs:177
        15: 0x55f4318cca9f - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h43f627be6f0625f9
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/panic/unwind_safe.rs:272
        16: 0x55f431895400 - std::panicking::try::do_call::h4a3e6561f81968e3
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/std/src/panicking.rs:554
        17: 0x55f4318954db - __rust_try
        18: 0x55f4318952fe - std::panicking::try::hf6d12dcdabc42c73
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/std/src/panicking.rs:518
        19: 0x55f43186b4ae - std::panic::catch_unwind::h6ff33c0c2d87e1db
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/std/src/panic.rs:142
        20: 0x55f4318d2f08 - nu_cli::repl::evaluate_repl::h04545f097074b6d1
                      at /home/fdncred/src/nushell/crates/nu-cli/src/repl.rs:176
        21: 0x55f4301a7edd - nu::run::run_repl::hc54f32f6d4bd8d15
                      at /home/fdncred/src/nushell/src/run.rs:278
        22: 0x55f43017947e - nu::main::h97b30ea9f7df9452
                      at /home/fdncred/src/nushell/src/main.rs:473
        23: 0x55f4301b556b - core::ops::function::FnOnce::call_once::h32d83572571d3fd2
                      at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ops/function.rs:250
  help: set the `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error:   × Main thread panicked.
  ├─▶ at crates/nu-protocol/src/engine/engine_state.rs:1050:14
zsh: terminated  RUST_BACKTRACE=full cargo r
```

### How to reproduce

see above

### Expected behavior

no crash

### Configuration

latest main, can't run to get the version

### Additional context

Since it's complaining about engine_state.rs:1050:14 It could be related to https://github.com/nushell/nushell/pull/13102/files#diff-4079a4a69643ccd80450ad15bffc099e28d55930f1a34118f65c7f005fad4014R1044

**Comment by NotTheDr01ds** [comment]
Same for me on WSL/Ubuntu 22.04/Nix-Unstable/Rustup.

Can confirm that #13120 fixes it for me

**Comment by kubouch** [comment]
Still not sure what could be the root cause, but I found something interesting. If you edit the `crates/nu-cli/src/reedline_config.rs` file (e.g., add a newline, whatever), then run `git diff` using [difftastic](https://github.com/Wilfred/difftastic), you'll see something strange:
![difft_menus](https://github.com/nushell/nushell/assets/25571562/a5635bad-5a5a-47df-bf61-690e94a17ec8)

Since there seems to be no change between the two, it looks like the newlines changed inside the raw strings. Maybe LF changed to CRLF? Evaluating these pieces of code was exactly what was causing the panic [here](https://github.com/nushell/nushell/blob/af22bb8d52a3e011cff9f0fb5766749206822dc3/crates/nu-cli/src/reedline_config.rs#L116). So, maybe the line endings were some span mismatch, although I still don't understand why things were working before without merging the delta...

**Comment by fdncred** [comment]
Line endings? That's weird. I merged #13120 on top of my latest PR and it's all working good now.

**Comment by kubouch** [comment]
It's weird, but looks at the diff... It definitely has something to do with line endings...
